java/cdk/src/main/java/sleeper/cdk/stack/CompactionStack.java

```diff
@@ -488,6 +494,14 @@ private void addEC2CapacityProvider(
         instanceProperties.set(COMPACTION_AUTO_SCALING_GROUP, ec2scalingGroup.getAutoScalingGroupName());
     }
 
+    private static List<ISecurityGroup> addSecurityGroupReferences(Construct scope, InstanceProperties instanceProperties) {
+        AtomicInteger index = new AtomicInteger(1);
+        return instanceProperties.getList(ECS_SECURITY_GROUPS).stream()
+                .filter(Predicate.not(String::isBlank))
+                .map(groupId -> SecurityGroup.fromSecurityGroupId(scope, "CompactionScalingSG" + index.getAndIncrement(), groupId))
```

Comment:
The documentation seems to suggest that if you use `SecurityGroup.fromSecurityGroupId`, it will add rules to the security group for ingress. I'm not sure what it means by that, or at what point it would add those rules, but it seems worrying that it suggests it will modify the group.

There's also `SecurityGroup.fromLookupById`, which says it'll just look it up. It seems a bit bizarre that this is different though.